storage/pebble/registers.go

```diff
@@ -0,0 +1,40 @@
+package pebble
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/cockroachdb/pebble"
+	"go.uber.org/multierr"
+
+	"github.com/onflow/flow-go/storage/pebble/registers/payload"
+)
+
+// library that implements pebble storage for registers
+type Registers struct {
+	Payloads *payload.Payloads
+}
+
+func StoragePath(dir string) string {
+	return path.Join(dir, "payload.db")
+}
+
+func NewRegisters(dir string, blockCacheSize int64) (*Registers, error) {
+	cache := pebble.NewCache(blockCacheSize)
+	defer cache.Unref()
```

Comment:
is this safe? The [docs](https://github.com/cockroachdb/pebble/blob/c872ea071a9f1eeb3b3fed09fdb1bb028ff5f6fb/internal/cache/clockpro.go#L783-L784) say
> The cache only remains valid as long a reference is maintained to it.

so it seems like the cache would be immediately invalidated when returning from this method.

Reply:
This is an optimization from @SaveTheRbtz I think

Reply:
[It should be fine](https://pkg.go.dev/github.com/cockroachdb/pebble#NewCache):

>  Each DB it is associated with adds a reference, so the creator of the cache should usually release their reference after the DB is created.

Reply:
ah, thanks!

Reply:
wow I didn't think summoning him would work!